### ISSUE TITLE
`prefer-tag-over-role` causing warnings on valid usage

### ISSUE THREAD

**Issue by geotrev** [jsx-eslint/eslint-plugin-jsx-a11y#920]
I've been updating a few pieces of code after updating to include this rule, and I'm finding that certain usages are flagged possibly incorrectly?

For e.g., I have code like so that is used as a global page alert element:

```html
<div role='status'>...</div>
<div role='alert'>...</div>
<svg role='img'>...</svg>
```

And gives the following output:

```
Use <output> instead of the ["status"/"alert"] role to ensure accessibility across all devices.
```

Which is strange because `<output>` is [primarily used in forms](https://www.w3schools.com/tags/tag_output.asp), unlike alert and status containers which which are often treated as live regions or focus trapped containers.

I did some digging and the both [live regions](https://www.digitala11y.com/status-role/#:~:text=The%20status%20role%20is%20used,be%20announced%20to%20the%20user.) and [SVG images](https://dequeuniversity.com/rules/axe/4.2/svg-img-alt) tends to be valid and recommended in these contexts.

**Comment by ljharb** [comment]
I'd suggest never referring to w3schools - it's not a good resource.

MDN doesn't indicate in any way that it's primarily for forms: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/output Certainly there's a `form` property but the "if any" means it's intended for use outside of forms as well.

**Comment by geotrev** [comment]
@ljharb Regardless, I still think the `div`/`role` usage being flagged is valid in this case given existing pattern documentation?

**Comment by ljharb** [comment]
I've tagged in @jessebeach on that one :-)
